src/index.js

```diff
@@ -59,6 +62,16 @@ const mainHandler = () => {
   checkAndUnchecklist(todos, localStorageTodos);
 
   editTodo(localStorageTodos);
+
+    // Delete a single todo
+  document.querySelectorAll('.delete-todo').forEach((list) => {
+    list.addEventListener('click', (e) => {
+      const todos = JSON.parse(localStorage.getItem(localStorageTodos));
+      const filtered = todos.filter((list) => list.id !== e.target.parentNode.dataset.id);
+      localStorage.setItem(localStorageTodos, JSON.stringify(filtered));
+      e.target.closest('ul>li').remove(todos);
+    });
+  });
 };
```

Comment:
- [ ] good job adding the delete function to your application 👍 however it doesn't quite work as it is supposed to, when the user deletes a todo item it disappears however when after that the user enters a new todo item the one which they deleted is added to the list again as well. it seems this is due to the fact that you are updating the local storage however you are not updating the `todos` constant.